src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -218,6 +220,11 @@
     public static final String MSG_MULTIPLE_LINES_INSIDE =
             "empty.line.separator.multiple.lines.inside";
 
+    /** List of AST token types, which can not have comment nodes inside. */
+    private static final List<Integer> TOKEN_TYPES_WITHOUT_COMMENTS_INSIDE =
+            Arrays.asList(TokenTypes.PACKAGE_DEF, TokenTypes.IMPORT, TokenTypes.STATIC_IMPORT,
```

Comment:
> List of AST token types, which can not have comment nodes inside

Based on the description alone, this doesn't seem right.
Packages and imports can have comments inside them.
Example:
````
package a/*a*/./*b*/b;
````
````
PACKAGE_DEF -> package [1:0]
|--ANNOTATIONS -> ANNOTATIONS [1:14]
|--BLOCK_COMMENT_BEGIN -> /* [1:9]
|  |--COMMENT_CONTENT -> a [1:11]
|  `--BLOCK_COMMENT_END -> */ [1:11]
|--DOT -> . [1:14]
|  |--IDENT -> a [1:8]
|  |--BLOCK_COMMENT_BEGIN -> /* [1:15]
|  |  |--COMMENT_CONTENT -> b [1:17]
|  |  `--BLOCK_COMMENT_END -> */ [1:17]
|  `--IDENT -> b [1:20]
`--SEMI -> ; [1:21]
````

Reply:
renamed